### ISSUE TITLE
the API loss

### ISSUE THREAD

**Issue by Moqixis** [Karel911/TRACER#33]
Hi, sorry to bother you. I use my  own dataset to train the model and I have faced the same trouble like https://github.com/Karel911/TRACER/issues/9
and I found the "amae" in the TRACER/util/losses.py was so big. Is this normal? 😔

Your code is
```
def adaptive_pixel_intensity_loss(pred, mask):
    w1 = torch.abs(F.avg_pool2d(mask, kernel_size=3, stride=1, padding=1) - mask)
    w2 = torch.abs(F.avg_pool2d(mask, kernel_size=15, stride=1, padding=7) - mask)
    w3 = torch.abs(F.avg_pool2d(mask, kernel_size=31, stride=1, padding=15) - mask)

    omega = 1 + 0.5 * (w1 + w2 + w3) * mask

    bce = F.binary_cross_entropy(pred, mask, reduce=None)
    abce = (omega * bce).sum(dim=(2, 3)) / (omega + 0.5).sum(dim=(2, 3))

    inter = ((pred * mask) * omega).sum(dim=(2, 3))
    union = ((pred + mask) * omega).sum(dim=(2, 3))
    aiou = 1 - (inter + 1) / (union - inter + 1)

    mae = F.l1_loss(pred, mask, reduce=None)
    amae = (omega * mae).sum(dim=(2, 3)) / (omega - 1).sum(dim=(2, 3))

    return (0.7 * abce + 0.7 * aiou + 0.7 * amae).mean()
```

In your article: 
![image](https://user-images.githubusercontent.com/44128124/210790480-0afb647c-7fd6-42c1-bdbc-0620b56eac7f.png)

I want to know in your code, how do you use H*W in the denominator in Eq. 11?
Thank you very much and look forward to your reply!!

**Comment by Karel911** [comment]
Sorry for late reply.

If you observed the amae occupying most of loss, there are two options.

Use a normal mae or drop amae.

If there is no improvement despite the modification, plz let me know.

**Comment by Moqixis** [comment]
> Sorry for late reply.
> 
> If you observed the amae occupying most of loss, there are two options.
> 
> Use a normal mae or drop amae.
> 
> If there is no improvement despite the modification, plz let me know.

I get it! Thank you!!!

**Comment by Moqixis** [comment]
Sorry to bother you, I have known the answer of
Q: I want to know in your code, how do you use H*W in the denominator in Eq. 11?

H*W have been calculated in _"mae = F.l1_loss(pred, mask, reduce=None)"_

Thank you very much for your great work!

**Comment by Karel911** [comment]
Please refer to the following lines [39 and 49](https://github.com/Karel911/TRACER/blob/main/util/effi_utils.py#L39).
